frontend/src/components/Banner.tsx

```diff
@@ -0,0 +1,90 @@
+import { ReactNode } from "react";
+import { OutboundLink } from "react-ga";
+import { useLocalization } from "@fluent/react";
+import styles from "./Banner.module.scss";
+import warningIcon from "../../../static/images/icon-orange-info.svg";
+import { useLocalDismissal } from "../hooks/localDismissal";
+import { CloseIcon } from "./icons/close";
+import { useGaPing } from "../hooks/gaPing";
+
+export type BannerProps = {
+  children: ReactNode;
+  type?: "promo" | "warning";
+  title?: string;
+  illustration?: ReactNode;
+  cta?: {
+    target: string;
+    content: string;
+    onClick?: () => void;
+    gaPing?: Parameters<typeof useGaPing>[0];
```

Comment:
I mostly went with this to indicate that this function blindly passes on the value of this property to `useGaPing`, so e.g. if `useGaPing` updates its first parameter, then whatever will be newly possible will also be possible here.

I think the sniff is because this is basically type-level programming, i.e. the type is not just a name, but has to be interpreted (both by the compiler and by the person reading this). Partly this is probably the result of TypeScript being a structural type system rather than nominal, but I'm sympathetic that it raises the cognitive barrier, especially if the person who reads this doesn't commonly read TypeScript. So if you feel strongly about this, or others stumble as well, I can also change this to `GaPingArgs | null`.